plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/TestKafkaSslConfig.java

```diff
@@ -0,0 +1,157 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.kafka.security.KafkaEndpointIdentificationAlgorithm;
+import io.trino.plugin.kafka.security.KafkaKeystoreTruststoreType;
+import io.trino.plugin.kafka.security.KafkaSslConfig;
+import org.testng.annotations.Test;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
+import static io.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
+import static io.airlift.configuration.testing.ConfigAssertions.recordDefaults;
+import static org.apache.kafka.common.config.SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_KEYSTORE_TYPE_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_KEY_PASSWORD_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG;
+import static org.apache.kafka.common.config.SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestKafkaSslConfig
+{
+    @Test
+    public void testDefaults()
+    {
+        // No defaults, should be empty if nothing is set specifically
+        assertRecordedDefaults(recordDefaults(KafkaSslConfig.class)
+                .setKeystoreLocation(null)
+                .setKeystorePassword(null)
+                .setKeystoreType(null)
+                .setKeyPassword(null)
+                .setTruststoreLocation(null)
+                .setTruststorePassword(null)
+                .setTruststoreType(null)
+                .setEndpointIdentificationAlgorithm(null));
+    }
+
+    @Test
+    public void testExplicitPropertyMappings()
+            throws IOException
+    {
+        String secret = "confluent";
+        Path keystorePath = Files.createTempFile("keystore", ".jks");
+        Path truststorePath = Files.createTempFile("truststore", ".jks");
+
+        createTempFile(keystorePath, secret);
+        createTempFile(truststorePath, secret);
```

Comment:
it does not have to be a real file here.

Reply:
I'm not sure I understand. The files are created with `Files.createTempFile()`

Reply:
It seems like it does require a real file to be here after we added the annotation `@FileExists`. I have commented it out and got an exception:

```
com.google.inject.ConfigurationException: Guice configuration errors:

1) Invalid configuration property kafka.ssl.keystore.location: file does not exist: /a/file (for class io.trino.plugin.kafka.security.KafkaSslConfig.keystoreLocation)

2) Invalid configuration property kafka.ssl.truststore.location: file does not exist: /a/file (for class io.trino.plugin.kafka.security.KafkaSslConfig.truststoreLocation)

2 errors

	at io.airlift.configuration.Problems.getException(Problems.java:135)
	at io.airlift.configuration.Problems.throwIfHasErrors(Problems.java:73)
	at io.airlift.configuration.ConfigurationFactory.build(ConfigurationFactory.java:411)
	at io.airlift.configuration.ConfigurationFactory.build(ConfigurationFactory.java:299)
	at io.airlift.configuration.ConfigurationFactory.build(ConfigurationFactory.java:294)
	at io.airlift.configuration.testing.ConfigAssertions.newInstance(ConfigAssertions.java:347)
	at io.airlift.configuration.testing.ConfigAssertions.assertFullMapping(ConfigAssertions.java:130)
	at io.trino.plugin.kafka.TestKafkaSslConfig.testExplicitPropertyMappings(TestKafkaSslConfig.java:90)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
```

Reply:
I think the comment was about the content of the file which is not important here.